llvm/lib/Transforms/Vectorize/LoopVectorizationPlanner.h

```diff
@@ -479,7 +479,11 @@ class LoopVectorizationPlanner {
   /// Build VPlans for the specified \p UserVF and \p UserIC if they are
   /// non-zero or all applicable candidate VFs otherwise. If vectorization and
   /// interleaving should be avoided up-front, no plans are generated.
-  void plan(ElementCount UserVF, unsigned UserIC);
+  /// DiffChecks is a list of pointer pairs that should be checked for aliasing,
```

Comment:
Better to add the new recipes at the same place where the runtime checks are attached to VPlan, rather than threading through the general planning stage: https://github.com/llvm/llvm-project/blob/main/llvm/lib/Transforms/Vectorize/LoopVectorize.cpp#L9359C32-L9359C51

Reply:
Yep that looks like a good first step. Would be good if updating the masks could also be done there (the PopCount introduction) and we could avoid making `addActiveLaneMask` more complex